lib/index.js

```diff
@@ -31,12 +66,12 @@ export default class Pageres {
 	 * @api public
 	 */
 
-	src(url, sizes, options) {
+	// src(url?: string, sizes?: Array<string>, options?: Options): Pageres | Array<Src> {
+	src(url, sizes, options): Pageres | Array<Src> { // TODO: how can I make all the arguments optional or all required? the commented out line doesn't work as it complains it can also be `undefined`. Can I define a separate type definiton for each scenario?
```

Comment:
This is the best way to do intersection types on classes:

```js
type getOrSet<Value> =
  & ((name: string, _: void) => Value)       // _ is necessary so undefined `Value`'s don't end up getting matched here
  & ((name: string, value: Value) => void);

class Foo<Value> {
  getOrSet: getOrSet<Value>; // overrides whatever `getOrSet` actually is because you can't do intersections otherwise
  getOrSet(name, value) {
    // ...
  }
}

let foo = new Foo();

(foo.getOrSet('bar', 2): void);
(foo.getOrSet('bar'): 2);
```